Makefile

```diff
@@ -60,89 +56,105 @@ ifneq ($(shell command -v dpkg),)
 	endif
 endif
 
-OS := $(shell go env GOOS)
-ARCH := $(shell go env GOARCH)
-
-kepler: build-containerized-cross-build-linux-amd64
-.PHONY: kepler
-
-tidy-vendor:
-	go mod tidy
-	go mod vendor
+GOOS := $(shell go env GOOS)
+GOARCH := $(shell go env GOARCH)
 
+# for testsuite
+ENVTEST_ASSETS_DIR=./test-bin
+export PATH := $(PATH):./test-bin
 
-_build_local: format
-	@mkdir -p "$(CROSS_BUILD_BINDIR)/$(GOOS)_$(GOARCH)"
-	+@GOOS=$(GOOS) GOARCH=$(GOARCH) go build -tags ${GO_BUILD_TAGS} \
-		-o $(CROSS_BUILD_BINDIR)/$(GOOS)_$(GOARCH)/kepler -ldflags $(LDFLAGS) ./cmd/exporter.go
+ifndef GOPATH
+  GOPATH := $(HOME)/go
+  GOBIN := $(GOPATH)/bin
+endif
 
-cross-build-linux-amd64: genbpfassets
-	+$(MAKE) _build_local GOOS=linux GOARCH=amd64
-.PHONY: cross-build-linux-amd64
+KUSTOMIZE = $(shell pwd)/bin/kustomize
 
-cross-build-linux-arm64: genbpfassets
-	+$(MAKE) _build_local GOOS=linux GOARCH=arm64
-.PHONY: cross-build-linux-arm64
+# go-get-tool will 'go get' any package $2 and install it to $1.
+PROJECT_DIR := $(shell dirname $(abspath $(firstword $(MAKEFILE_LIST))))
+define go-get-tool
+@[ -f $(1) ] || { \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_DIR)/bin go install $(2) ;\
+ls $$TMP_DIR;\
+echo $(PROJECT_DIR);\
+rm -rf $$TMP_DIR ;\
+}
+endef
 
-cross-build-linux-s390x: genbpfassets
-	+$(MAKE) _build_local GOOS=linux GOARCH=s390x
-.PHONY: cross-build-linux-s390x
+base_dir := $(patsubst %/,%,$(dir $(realpath $(lastword $(MAKEFILE_LIST)))))
 
-cross-build: cross-build-linux-amd64 cross-build-linux-arm64 cross-build-linux-s390x
-.PHONY: cross-build
+### Default ###
+kepler: _build_containerized
+.PHONY: kepler
 
+clean: clean-cross-build
+.PHONY: clean
 
+### build container ###
 _build_containerized: genbpfassets tidy-vendor format
 	@if [ -z '$(CTR_CMD)' ] ; then echo '!! ERROR: containerized builds require podman||docker CLI, none found $$PATH' >&2 && exit 1; fi
 	echo BIN_TIMESTAMP==$(BIN_TIMESTAMP)
 
-	$(CTR_CMD) build -t $(IMAGE_REPO)/kepler:$(SOURCE_GIT_TAG)-linux-$(ARCH) \
+	$(CTR_CMD) build -t $(IMAGE_REPO)/kepler:$(SOURCE_GIT_TAG)-linux-$(GOARCH) \
 		-f "$(SRC_ROOT)"/build/Dockerfile \
 		--build-arg SOURCE_GIT_TAG=$(SOURCE_GIT_TAG) \
 		--build-arg BIN_TIMESTAMP=$(BIN_TIMESTAMP) \
-		--build-arg ARCH=$(ARCH) \
-		--build-arg MAKE_TARGET="cross-build-linux-$(ARCH)" \
-		--platform="linux/$(ARCH)" \
+		--platform="linux/$(GOARCH)" \
 		.
 
-	$(CTR_CMD) tag $(IMAGE_REPO)/kepler:$(SOURCE_GIT_TAG)-linux-$(ARCH) $(IMAGE_REPO)/kepler:$(IMAGE_TAG)
+	$(CTR_CMD) tag $(IMAGE_REPO)/kepler:$(SOURCE_GIT_TAG)-linux-$(GOARCH) $(IMAGE_REPO)/kepler:$(IMAGE_TAG)
 
 .PHONY: _build_containerized
 
-build-containerized-cross-build-linux-amd64:
```

Comment:
I don't get the reason of the design here....
if we use buildx, then for each platform it able to get OS/CPU arch by GOENV, which means `make _build_containerized` works for all platforms. Furthermore, without buildx support, we can't assume use CPU in arch A to build arch B as the base image are different. Right?